lib/util/source-code.js

```diff
@@ -138,7 +139,26 @@ function SourceCode(text, ast) {
      * This is done to avoid each rule needing to do so separately.
      * @type string[]
      */
-    this.lines = SourceCode.splitLines(this.text);
+    this.lines = [];
+    this.lineStartIndices = [0];
+
+    const lineEndingPattern = astUtils.createGlobalLinebreakMatcher();
```

Comment:
I agree that `SourceCode.splitLines` is now obsolete. I kept it because I wasn't sure whether removing it would be a breaking change.

It doesn't seem to be documented, but given that `SourceCode` is exposed on the `eslint` object, I'm wondering if it would be better to wait until 4.0.0 to remove it.